### PR TITLE
Generates ores in ChunkGenerator

### PR DESCRIPTION
**Pull request by tastybento**
Blocks cannot be set in populators so no ores were being generated. So I
moved ore generstion into the chunk generator.

WIP: Also, all entities (except Shulkers) are dying instantly because blocks are not being set to air in the entity populator. This needs fixing still.

### CODE REVIEW

**Comment by tastybento** [comment]
Let me try and fix the entity population before merging.

**Comment by tastybento** [comment]
Okay, this is weird. On further investigation, block populators should be able to set the block type. This is not happening for some reason.

**Comment by tastybento** [comment]
So, I'm not sure why your block populators were not actually setting blocks (or maybe they were?) but anyway, this latest commit works. The problem is that the server will despawn living entities not close to the player, so the result is that there are quite a few empty spaces with nothing in them (or water). So, I'm not so sure how useful having living entity population.

**Comment by tastybento** [comment]
Okay, final comment - I'm not sure if this is exactly what you wanted, but I hope it's useful. It's been quite a trip trying to work all this out!

**Comment by BONNe** [comment]
Its weird...for me all populators worked. 
And for shulkers also was air block over it. 

Other mobs were not done, as it is usesless. Most of them will despawn.

File: src/main/resources/config.yml [hunk 1]
```diff
@@ -88,6 +86,9 @@ world:
     - MATERIAL:GRANITE:20:10
     - MATERIAL:ANDESITE:20:10
     - MATERIAL:DIORITE:30:8
+    - ENTITY:ZOMBIE:1:1
+    - ENTITY:DROWNED:1:1
+    - ENTITY:TROPICAL_FISH:1:1
```
Comment (BONNe) [inline src/main/resources/config.yml]: All these mobs will be despawned as soon as there will be no players around.

File: src/main/java/world/bentobox/caveblock/generators/populators/EntitiesPopulator.java [hunk 2]
```diff
@@ -270,6 +153,17 @@ private void tryToPlaceEntity(World world, Block block, EntityType entity, int x
...
+    List<Block> space = new ArrayList<>();
+
+    private final static List<EntityType> WATER_ENTITIES = Arrays.asList(EntityType.GUARDIAN,
+            EntityType.SQUID,
+            EntityType.COD,
+            EntityType.SALMON,
+            EntityType.PUFFERFISH,
+            EntityType.TROPICAL_FISH,
+            EntityType.DROWNED,
+            EntityType.DOLPHIN);
```
Comment (BONNe) [inline src/main/java/world/bentobox/caveblock/generators/populators/EntitiesPopulator.java]: Dolphin needs air block above it, otherwise he will die.

**Comment by BONNe** [comment]
> Blocks cannot be set in populators so no ores were being generated. So I
> moved ore generstion into the chunk generator.

So, in first try I also done it in generator, but it is not a proper way how to do it. Correct way is using BlockPopulators.

**Comment by tastybento** [comment]
Okay, that's embarrassing. Yes, ores are being generated. I think the issue was that I was changing the config file setting to increase the ore generation rate while the server was running without doing /cbadmin reload. Ugh.

**Pull request closed without merging**